pkg/services/dashboards/folder_service.go

```diff
@@ -96,6 +97,25 @@ func (dr *dashboardServiceImpl) GetFolderByUID(uid string) (*models.Folder, erro
 	return dashToFolder(dashFolder), nil
 }
 
+func (dr *dashboardServiceImpl) GetFolderBySlug(slug string) (*models.Folder, error) {
+	query := models.GetDashboardQuery{OrgId: dr.orgId, Slug: slug}
+	dashFolder, err := getFolder(query)
+
+	if err != nil {
```

Comment:
Just a nit, but preferable for the reader if error checks are co-located with the error assignment.
```suggestion
	dashFolder, err := getFolder(query)
	if err != nil {
```